MDBOOK: Harvest FAQ from proto into MDBOOK -329

murrayjbrown opened the pull request:
@Connoropolous

Connoropolous commented:
Ah, oh wow @murrayjbrown ... this is intense. Is this content from throughout the wiki?

murrayjbrown commented:
@Connoropolous It's the FAQ page from the wiki. From the way the issues are worded, I presumed that you wanted the FAQ as a separate MDBOOK, whereas for #330 the rest of the wiki content is to be migrated into the _holochain_101_ MDBOOK. Is that right?

Connoropolous commented:
I wasn't thinking strictly of doing it separately, no, I hadn't considered it actually. I think it could be integrated? What do you think?

murrayjbrown commented:
It's certainly useful to include the FAQ within the _holochain_101_ MDBOOK. There may or may not be utility in packaging a separate FAQ MDBOOK. So, was the separation of #329 and #330 more a staging matter? I can do it either way.

murrayjbrown commented:
D'oh, looking at it again, now, I see that I'd neglected to remove the unreferenced wiki files.

The landing page is _Home.md_ and the FAQ is _FAQ.md_ and they share some included files but most are referenced by _Home.md._ Where did you intend the non-FAQ wiki content to be published? I'd thought parts of it was to be folded into the _holochain_101_ MDBOOK, as and where appropriate.

Connoropolous commented:
To your last point, precisely. In one sense, it would be best to be pulled out and into different areas where appropriate. That part could wait for a little further thought to go into it.

murrayjbrown commented:
Okay, good. So let me go ahead and move the FAQ into the _holochain_101_ MDBOOK for issue #329.

Connoropolous commented:
@murrayjbrown so is this good for review again?

murrayjbrown commented:
Yes. It will need some content revisions, though, as noted by TODOs in the document, when updated information is available.

Connoropolous commented on doc/holochain_101/src/FAQ.md:
```diff
@@ -0,0 +1,240 @@
+
+Please see the [Comparisons page](https://github.com/Holochain/holochain-proto/wiki/Comparisons).
+
+# Who is on the development team? What is their history and experience?
```
Let's take this section out from here entirely. The right place for that info will be somewhere else.

murrayjbrown replied:
@Connoropolous Sure. Just the Comparisons section?

Connoropolous replied:
Nah, the "who is on the dev team / history /experience"

murrayjbrown replied:
@Connoropolous Okay, will do.

murrayjbrown replied:
@Connoropolous BTW, I can add navigation next/previous page links to the MDBOOK via a simple build option, if you like.

zippy replied:
agreed re removing team.  Out of date and wrong place...  That was more relevant back in ICO days.

murrayjbrown commented:
@zippy Will do.

murrayjbrown commented:
@Connoropolous @zippy Requested changes ready for review. I merged latest commits, resolving a flagged conflict, fixed some links, FAQ header levels, and added that MDBOOK build option for previous/next page links.

Connoropolous commented on doc/holochain_101/book.toml:
```diff
@@ -3,3 +3,6 @@ authors = ["David Meister"]
 title = "holochain_101"
+
+[build]
+preprocess = ["links"]
```
what does this do?

murrayjbrown replied:
@Connoropolous It adds the prev/next page links to the MDBOOK.

Connoropolous replied:
Cool, I like it

Connoropolous commented on README.md:
```diff
@@ -12,7 +12,7 @@ This is the home of the Holochain Rust library, being rewritten from [Go](https:
 <br/>
 
-| Holochain Links: | [FAQ](https://github.com/holochain/holochain-proto/wiki/FAQ) | [Developer Docs](https://holochain.github.io/holochain-rust/) | [White Paper](https://github.com/holochain/holochain-proto/blob/whitepaper/holochain.pdf) |
+| Holochain Links: | [FAQ](https://github.com/holochain/holochain-rust/wiki/FAQ) | [Developer Docs](https://holochain.github.io/holochain-rust/) | [White Paper](https://github.com/holochain/holochain-proto/blob/whitepaper/holochain.pdf) |
```
won't it be: https://holochain.github.io/holochain-rust/faq.html

murrayjbrown replied:
@Connoropolous  Sorry, yeah, I just realized myself that a simple substitution wasn't right. Your link is close but the _FAQ.html_ base name is uppercase (as was the original); I'll change that to lowercase, if you prefer. Probably better that way.

murrayjbrown replied:
@Connoropolous Okay, I've pushed that change, with the FAQ in a renamed _faq.md_ file and links updated accordingly.
